parquet-column/src/main/java/org/apache/parquet/filter2/predicate/Statistics.java

```diff
@@ -26,17 +28,31 @@
 public class Statistics<T> {
   private final T min;
   private final T max;
+  private final Comparator<T> comparator;
 
-  public Statistics(T min, T max) {
+  public Statistics(T min, T max, Comparator<T> comparator) {
     this.min = checkNotNull(min, "min");
     this.max = checkNotNull(max, "max");
+    this.comparator = checkNotNull(comparator, "comparator");
   }
 
+  /**
+   * The self-comparison logic of {@code T} might not proper for the actual logical type (e.g. unsigned int). Use {@link
```

Comment:
This is a useful note, but not javadoc for this method. If you want to add Javadoc here, describe what the method does and then add a note below that. Same for `getMax`.